include/dlaf/init.h

```diff
@@ -0,0 +1,114 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2020, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+#pragma once
+
+#include <hpx/execution.hpp>
+#include <hpx/functional.hpp>
+
+#include <dlaf/types.h>
+
+#ifdef DLAF_WITH_CUDA
+#include <dlaf/cublas/executor.h>
+#endif
+
+namespace dlaf {
+namespace internal {
+bool& initialized();
+
+template <Backend D>
+struct Init {
+  // Initialization and finalization does nothing by default. Behaviour can be
+  // overridden for backends.
+  static void initialize(int, char**) {}
+  static void finalize() {}
```

Comment:
We can also explicitly make the `CPU` backend do nothing. However, I thought this was appropriate default behaviour.